test_scripts/AppIconsFolder/ATF_N_AppIconsAmountToRemove_larger_than_icons_amount.lua

```diff
@@ -0,0 +1,245 @@
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test.Precondition_stopSDL()
+  StopSDL()
+end 
+
+function Test.Precondition_configureAppIconsFolder()
+  commonFunctions:SetValuesInIniFile("AppIconsFolder%s-=%s-.-%s-\n", "AppIconsFolder", 'Icons')
+end
+ 
+function Test.Precondition_configureAppIconsFolderMaxSize()
+  commonFunctions:SetValuesInIniFile("AppIconsFolderMaxSize%s-=%s-.-%s-\n", "AppIconsFolderMaxSize", 1048576)
+end
+
+function Test.Precondition_configureAppIconsAmountToRemove()
+  commonFunctions:SetValuesInIniFile("AppIconsAmountToRemove%s-=%s-.-%s-\n", "AppIconsAmountToRemove", 100)
+end
+ 
+function Test.Precondition_removeAppIconsFolder()
+  local addedFolderInScript = "Icons"
+    local existsResult = commonSteps:Directory_exist(tostring(config.pathToSDL .. addedFolderInScript))
+  if existsResult == true then
+    local rmAppIconsFolder  = assert( os.execute( "rm -rf " .. tostring(config.pathToSDL .. addedFolderInScript)))
+    if rmAppIconsFolder ~= true then
+      commonFunctions:userPrint(31, tostring(addedFolderInScript) .. " folder is not deleted")
+    end
+  end
+end
+
+function Test.Precondition_startSDL()
+  StartSDL(config.pathToSDL, config.ExitOnCrash)
+end
+
+function Test:Precondition_initHMI()
+  self:initHMI()
+end
+
+function Test:Precondition_initHMIonReady()
+  self:initHMI_onReady()
+end
+
+function Test:Precondition_connectMobile()
+  self:connectMobile()
+end
+
+function Test.Precondition_makeAppIconsFolderFull()
+  makeAppIconsFolderFull( "Icons" )
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:Check_old_icons_deleted_AmountToRemove_greater_than_icons_amount()
+  self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  self.mobileSession.version = 4
+  self.mobileSession:StartService(7)
+  :Do(function()
+    registerApplication(self)
+    EXPECT_NOTIFICATION("OnHMIStatus", { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE"})
+    :Do(function()
+     local cidPutFile = self.mobileSession:SendRPC("PutFile",
+      {
+        syncFileName = "iconFirstApp.png",
+        fileType = "GRAPHIC_PNG",
+        persistentFile = false,
+        systemFile = false
+      }, "files/icon.png")
+     EXPECT_RESPONSE(cidPutFile, { success = true, resultCode = "SUCCESS" })
+     :Do(function()
+     local cidSetAppIcon = self.mobileSession:SendRPC("SetAppIcon",{ syncFileName = "iconFirstApp.png" })
+     pathToAppFolder = pathToAppFolderFunction(RAIParameters.appID)
+     EXPECT_HMICALL("UI.SetAppIcon",
+      {
+        syncFileName =
+          {
+            imageType = "DYNAMIC",
+            value = pathToAppFolder .. "iconFirstApp.png"
+          }
+       })
+    :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    EXPECT_RESPONSE(cidSetAppIcon, { resultCode = "SUCCESS", success = true })
+    :ValidIf(function()
+      checkFunction()
```

Comment:
@aarotar, ValidIf() function should return true or false as 1st parameter and reason of fail as the 2nd. Therefore:
1) Code of ValidIf() has to be modified in this way
2) Code of checkFunction() has to be modified in a way that it returns 2 values: 
- status of the check
- reason of fail